module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,59 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+pairVector compressGrayscale(const twoDimensionalArray& inputMap) {
+    pairVector outputVec;
+    outputVec.reserve(width * height);
+
+    std::all_of(inputMap.begin(), inputMap.end(),
+                [&outputVec](auto mapRow) {
+                    for (auto tempIt = mapRow.begin(); tempIt != mapRow.end(); ++tempIt) {
+                        auto it = std::adjacent_find(tempIt, mapRow.end(), std::not_equal_to<int>{});
+                        if (it == mapRow.end()) {
+                            it--;
+                        }
+                        outputVec.emplace_back(std::make_pair(*it, (std::distance(tempIt, it) + 1)));
+                        tempIt = it;
+                    }
+
+                    return true;
```

Comment:
You're using all_of to create side effects (vec.emplace_back). That's generally a code smell - you should use for_each here. Algorithms that 'return' something (that is not iterator) should not have side effects.